### DISCUSSION TITLE
How to actually use locally-built Xtext?

### DISCUSSION THREAD

**Discussion by tfonda-fbk** [eclipse-xtext/xtext#3070] [Q&A] [answered]
Hi,

I've compiled Xtext on my machine but I'm having a hard time figuring out how to actually use the compiled artifacts. I've followed the instructions provided in `CONTRIBUTING.md` and completed section "Create a Release Locally" successfully. The documentation apparently goes no further than this point.

This is what I tried doing afterwards. In Eclipse I opened the "Install new software" window, added a new repository, clicked on Archive and chose the `org.eclipse.xtext.p2repository-2.35.0-SNAPSHOT.zip` zip file that had been generated in `build/`. In the following screen I updated the Xtext SDK to my locally-built version, which was suffixed with the correct date (this made me believe I was on the right track).

After restarting the IDE I created a new sample Xtext project and generated its artifacts. However, when trying to launch the sample Xtext project as an Eclipse application, I see this error message:

![Schermata_20240528_094553](https://github.com/eclipse/xtext/assets/167330712/a5e18eac-f5d4-4d3e-99d9-0f3180dc56df)

I am not sure what is causing this, since I performed a 100% vanilla build of Xtext (I have not modified a single line of code), and I am not sure how to proceed either.

Any insight?
Thanks in advance!

**Comment by LorenzoBettini** [comment]
Hi, that's something needed from the Eclipse LSP4E; you might want to add a recent Eclipse update and/or Orbit site so that you can install the missing feature.

On a side note, for quickly manually testing your modifications to Xtext, if you did modifications in a workspace created with Oomph, it should be enough to start a new Eclipse instance from there.

**Comment by tfonda-fbk** [comment]
Thanks for the quick reply. I believe I have installed gson correctly but the errors are still there.
![gson_installed](https://github.com/eclipse/xtext/assets/167330712/d47dd893-b7fb-4fe0-af5d-1b85c078fa27)

I've searched for gson jars in my home directory and found many:
```
lab@latitude7430:~> find . -type f -name "*gson*.jar"
./devel/xtext/ws/.metadata/.plugins/org.eclipse.pde.core/.bundle_pool/plugins/com.google.gson_2.10.1.jar
./devel/xtext/ws/.metadata/.plugins/org.eclipse.pde.core/.bundle_pool/plugins/com.google.gson_2.11.0.jar
./devel/xtext/ws/.metadata/.plugins/org.eclipse.pde.core/.bundle_pool/plugins/com.google.gson.source_2.11.0.jar
./devel/xtext/ws/.metadata/.plugins/org.eclipse.pde.core/.bundle_pool/plugins/com.google.gson.source_2.10.1.jar
./devel/xtext/git/xtext/org.eclipse.xtend.ide.tests/target/antDeps/gson-2.11.0.jar
./devel/xtext/git/xtext/org.eclipse.xtend.maven.plugin/target/.m2test/com/google/code/gson/gson/2.11.0/gson-2.11.0.jar
./devel/xtext/git/xtext/org.eclipse.xtext.maven.plugin/target/.m2test/com/google/code/gson/gson/2.11.0/gson-2.11.0.jar
./devel/xtext/git/xtext/org.eclipse.xtext.web.example.jetty/target/org.eclipse.xtext.web.example.jetty-2.35.0-SNAPSHOT/WEB-INF/lib/gson-2.11.0.jar
./.p2/pool/plugins/com.google.gson_2.10.1.jar
./.p2/pool/plugins/com.google.gson_2.10.1.v20230109-0753.jar
./.p2/pool/plugins/com.google.gson.source_2.10.1.jar
./.m2/repository/com/google/code/gson/gson/2.10.1/gson-2.10.1-sources.jar
./.m2/repository/com/google/code/gson/gson/2.10/gson-2.10.jar
./.m2/repository/com/google/code/gson/gson/2.11.0/gson-2.11.0.jar
./.m2/repository/com/google/code/gson/gson/2.6.2/gson-2.6.2.jar
./.m2/repository/.cache/tycho/https/download.eclipse.org/eclipse/updates/4.32-I-builds/I20240525-0900/plugins/com.google.gson_2.11.0.jar
./.m2/repository/.cache/tycho/https/download.eclipse.org/eclipse/updates/4.32-I-builds/I20240525-0900/plugins/com.google.gson.source_2.11.0.jar
./.m2/repository/.cache/tycho/https/download.eclipse.org/eclipse/updates/4.32-I-builds/I20240525-1800/plugins/com.google.gson.source_2.11.0.jar
./.m2/repository/p2/osgi/bundle/com.google.gson/2.11.0/com.google.gson-2.11.0.jar
./.m2/repository/p2/osgi/bundle/com.google.gson.source/2.11.0/com.google.gson.source-2.11.0.jar
```

I've opened a few of these jars and they indeed contain the com.google.gson.annotation package, as expected.
This however doesn't look like an Xtext-specific issue, so it's probably best if I ask for help somewhere else.

**Reply by LorenzoBettini** [reply]
> I've searched for gson jars in my home directory and found many:

Those results have nothing to do with your installed Eclipse: they've been using for building.

**Comment by LorenzoBettini** [comment]
Which version of Eclipse are you using?
For 2024-06 I think there a few dependencies that need to be aligned. See for example https://github.com/eclipse/xtext/issues/3060

**Comment by LorenzoBettini** [comment]
And as I said, try to run another Eclipse instance from the Xtext workspace, if you want to do some quick manual checks

**Comment by tfonda-fbk** [comment]
@LorenzoBettini I've set up the development environment using the Oomph automated configuration link. It installed Eclipse 2024-06. I am also seeing the issue with java processes using 100% of the CPU upon launching the DSL project, as you described in the issue you linked.
About running an Eclipse instance from the Xtext workspace: I'm not sure I understand this correctly, are you suggesting that I "Run as > Eclipse application" the org.eclipse.xtext project itself (the same way I'm used to run DSL projects)?
Thanks.

**Reply by LorenzoBettini** [reply]
> About running an Eclipse instance from the Xtext workspace: I'm not sure I understand this correctly, are you suggesting that I "Run as > Eclipse application" the org.eclipse.xtext project itself (the same way I'm used to run DSL projects)?

Exacty! That's standard of Eclipse plug-in development.

**Reply by tfonda-fbk** [reply]
@LorenzoBettini Okay, I'll do that from now on (but at the moment I of course see the same errors when launching it that way).

**Comment by LorenzoBettini** [comment, marked as answer]
See my temporary local workaround in the above linked issue
